pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -48,8 +49,9 @@ func NewFramework(r Registry, _ *runtime.Unknown) (Framework, error) {
 	// TODO: The framework needs to read the scheduler config and initialize only
 	// needed plugins. In this initial version of the code, we initialize all.
 	for name, factory := range r {
-		// TODO: 'nil' should be replaced by plugin config.
-		p, err := factory(nil, f)
+		// use configuration to decide which plugins to instantiate.
+		pc := pluginConfig(name, args)
```

Comment:
What if we put the plugin configs into a map so the value can be looked up directly for each factory?

Reply:
Updated. PTAL.
Thanks!!